homeassistant/components/tessie/sensor.py

```diff
@@ -0,0 +1,213 @@
+
+from .const import DOMAIN, TessieStatus
+from .coordinator import TessieDataUpdateCoordinator
+from .entity import TessieEntity
+
+PARALLEL_UPDATES = 0
+
+DESCRIPTIONS: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        key="state",
+        options=[status.value for status in TessieStatus],
+        device_class=SensorDeviceClass.ENUM,
+    ),
+    SensorEntityDescription(
+        key="charge_state-usable_battery_level",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=SensorDeviceClass.BATTERY,
+    ),
+    SensorEntityDescription(
+        key="charge_state-charge_energy_added",
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        suggested_display_precision=1,
+    ),
+    SensorEntityDescription(
+        key="charge_state-charger_power",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        device_class=SensorDeviceClass.POWER,
+    ),
+    SensorEntityDescription(
+        key="charge_state-charger_voltage",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfElectricPotential.VOLT,
+        device_class=SensorDeviceClass.VOLTAGE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="charge_state-charger_actual_current",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
+        device_class=SensorDeviceClass.CURRENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="charge_state-charge_rate",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfSpeed.MILES_PER_HOUR,
+        device_class=SensorDeviceClass.SPEED,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="charge_state-battery_range",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfLength.MILES,
+        device_class=SensorDeviceClass.DISTANCE,
+        suggested_display_precision=1,
+    ),
+    SensorEntityDescription(
+        key="drive_state-speed",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfSpeed.MILES_PER_HOUR,
+        device_class=SensorDeviceClass.SPEED,
+    ),
+    SensorEntityDescription(
+        key="drive_state-power",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        device_class=SensorDeviceClass.POWER,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="drive_state-shift_state",
+        icon="mdi:car-shift-pattern",
+        options=["P", "D", "R", "N"],
```

Comment:
Looks like these are upper but translations are lower?

Reply:
The native value is upper, but upper case characters are not allowed in translation keys I thought, ill go double check

Reply:
` Invalid translation key 'P', need to be [a-z0-9-_]+ `

Reply:
Need to check if we need to convert them to lower here or not or if the frontend auto lowers. I expect it does since it has a validator but I'm not 100% sure on that one

Reply:
I'll set up a tessie account and test as well

Reply:
If your gonna sign up: https://share.tessie.com/G7mQQoGR4Eb

Im going to test it now too.

Reply:
It doesnt work, ill lower case the values

Reply:
https://github.com/home-assistant/core/pull/104684#issuecomment-1847997535